app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,214 @@
+      @grouplessCUDs = @course.course_user_data.joins(:assessment_user_data).where(assessment_user_data: {assessment_id: @assessment.id, membership_status: AssessmentUserDatum::UNCONFIRMED})
+    else
+      aud = @assessment.aud_for @cud.id
+      if aud.group then
+        redirect_to [@course, @assessment, aud.group] and return
+      else
+        redirect_to action: :new and return
+      end
+    end
+    respond_with(@course, @assessment, @groups)
+  end
+
+  action_auth_level :show, :student
+  def show
+    @aud = @assessment.aud_for @cud.id
+    unless @cud.instructor? then
+      if @aud.group_id == nil then
+        redirect_to action: :new and return
+      elsif @aud.group_id != params[:id].to_i then
+        redirect_to [@course, @assessment, @aud.group] and return
+      end
+    end
+    if @group.size < @assessment.group_size then
+      @grouplessCUDs = @assessment.grouplessCUDs
+    end
+    respond_with(@course, @assessment, @group)
+  end
+
+  action_auth_level :new, :student
+  def new
+    @group = Group.new
+    @grouplessCUDs = @assessment.grouplessCUDs
+    @groups = @assessment.groups
+    respond_with(@course, @assessment, @group)
+  end
+
+  action_auth_level :create, :student
+  def create
+    if params[:member_id] then
+      if params[:member_id].to_i == @cud.id then
+        flash[:error] = "You can't create a group with just yourself"
+        redirect_to action: :new and return
+      end
+      @aud1 = @assessment.aud_for @cud.id
+      @aud2 = @assessment.aud_for params[:member_id].to_i
+      if @aud1.group_confirmed then
+        flash[:error] = "You already have a confirmed group."
+        redirect_to action: :new and return
+      elsif @aud2.group_confirmed then
+        flash[:error] = aud2.course_user_datum.email + " already has a confirmed group."
+        redirect_to action: :new and return
+      end
+      ActiveRecord::Base.transaction do
+        @group = Group.new
+        @group.name = "Untitled"
+        @group.save!
+        @aud1.group_id = @group.id
+        @aud1.membership_status = AssessmentUserDatum::CONFIRMED
+        @aud1.save!
+        @aud2.group_id = @group.id
+        @aud2.membership_status = AssessmentUserDatum::GROUP_CONFIRMED
+        @aud2.save!
+      end
+    end
+    respond_with(@course, @assessment, @group)
+  end
+
+  action_auth_level :update, :student
+  def update
+    if params[:group] then
+      flash[:notice] = 'Group was successfully updated.' if @group.update(group_params)
+    end
+    respond_with(@course, @assessment, @group)
+  end
+
+  action_auth_level :destroy, :student
+  def destroy
+    ActiveRecord::Base.transaction do
+      @group.assessment_user_data.each do |aud|
+        aud.group_id = nil
+        aud.membership_status = AssessmentUserDatum::UNCONFIRMED
+        aud.save!
+      end
+      @group.destroy!
+    end
+    respond_with(@course, @assessment, @group)
+  end
+ 
+  action_auth_level :import, :instructor
+  def import
```

Comment:
That's a feature that was previously available, and I guess this function will just try its best to make the groups look the same.